packages/flame/lib/src/game/camera/camera_wrapper.dart

```diff
@@ -0,0 +1,47 @@
+import 'dart:ui';
+import '../../../components.dart';
+import 'camera.dart';
+
+/// Do not use.
+///
+/// This class encapsulates BaseGame's rendering functionality. It will be
+/// converted into a proper Component in a future release, but until then
+/// using it in any code other than the BaseGame class is unsafe and
+/// not recommended.
+class CameraWrapper {
+  // TODO(st-pasha): extend from BaseComponent
+  CameraWrapper(this.camera, this.world);
+
+  final Camera camera;
+  final ComponentSet world;
+
+  void update(double dt) {
+    camera.update(dt);
+  }
+
+  void render(Canvas canvas) {
+    // TODO(st-pasha): it would be easier to keep the world and the
+    //   HUD as two separate component trees.
+    camera.viewport.render(canvas, (_canvas) {
+      // First render regular world objects
+      canvas.save();
+      camera.apply(canvas);
+      world.forEach((component) {
+        if (!component.isHud) {
+          canvas.save();
+          component.renderTree(canvas);
+          canvas.restore();
+        }
+      });
+      canvas.restore();
+      // Then render the HUD
```

Comment:
this is breaking the behaviour, currently components are drawn in priority order, regardless of isHUD!

Reply:
Right, I guess this could actually break components like the Parallax.
Since the parallax would be rendered on top of all the other components.

Reply:
Maybe HUD is actually a bad name for it? Since it isn't always really "heads-up".

Reply:
yeah the definition would be ignoreCamera or something. note that it still applies viewport. because that is how HUD works. but I could see you wanting ignoreViewport as well... I guess isHUD is just a nice way to encapsulate it for most use cases

Reply:
> currently components are drawn in priority order, regardless of isHUD!

That's true; but the current behavior is considered a bug (#934), so this PR actually fixes that bug. Should this perhaps be noted more explicitly somewhere?

Reply:
I guess that #934 is invalid (my fault), we should clean that up to remove confusion.
Would we want all three, `ignoreCamera`, `ignoreViewport` and `isHud` (having `isHud` working like it is in this branch) maybe?

Reply:
> Would we want all three, ignoreCamera, ignoreViewport and isHud maybe?

Ideally, we'd want neither. The semantics of these properties is too vague and error-prone. Consider the current `isHud` for example: at this moment this property works only for elements that are attached directly to the BaseGame class. If such elements appear at a more deep level in the components hierarchy, they would not be rendered correctly. This includes the current `ParallaxComponent` class too since apparently, it declares itself as `isHud`.

A better approach would be to model these properties via the components' place in the game tree:
```
Game
 +-- World (most of the in-game components go here)
 +-- Camera (views the World)
 |   +-- HUD (elements that are rendered on top of the camera's viewport)
 +-- Other (elements that are completely independent from the camera/viewport)
```

Reply:
I agree with this in parts. I think having the camera as a component that can have children makes a lot of sense. That allow for multiple cameras and hud elements can be out of the camera. however even hud elements should be subject to viewport. so I think it would look like this:

```
game (has a viewport)
         + -- "hud" components (not transformed by any camera)
         + -- camera 1 comp
         |        + -- components transformed by camera1
         + -- camera 2 comp
                  + -- components transformed by camera2
```

I don't think it makes any sense for a game to have more than one viewport. I don't see how that would apply (the viewport draws constraints on the screen, doesn't just translate stuff inside it).
so having that in mind I am not sure unifying camera and viewport would work for this structure. wdyt?

Reply:
cc @erickzanardo @spydon @wolfenrain @renancaraujo as I think this is an impactful decision

Reply:
A viewport is a rectangle (or some other shape) on the screen where a camera renders the world. The viewport's primary function is to clip the world (which is probably too large to fit on a screen anyway). The camera's function, on the other hand, is to move the world within the viewport, i.e. to choose the point within the world that the camera is currently "looking at". As such, a camera and a viewport are 2 parts of a single entity, and one cannot operate without the other.

For example, I'm currently making a demo game where there is a world that can be viewed normally on a screen, but also the ability to use a magnifying glass to look at some parts of the world more closely. The magnifying glass here is a separate camera (with its own custom viewport), which can be dynamically rendered on top of the main world. The magnifying glass even has some "hud" elements: the frame and the handle. These are bound to the viewport (i.e. move together with the viewport) but are not actually clipped by the viewport.

Reply:
How does that work when you are having the components affected by both cameras? Does it works because they are only seen through the last camera in the hierarchy?

Sounds to me like it make sense to have the viewport in the camera then. What about having the components that are affected by the camera as children to a camera component? I think we can disregard having two cameras on the same level with overlapping components for now. (The issue we talked about earlier)

Reply:
Yes, there are some components whose rendering logic might depend on the settings of the camera, such as zoom level, or the position of the viewport. For example, GoogleMaps gradually removes smaller details of the landscape as you zoom out. The Parallax is another example, where you have a large image and want to render only that portion of it which is visible through the viewport. In cases like these, when there are two cameras, the elements will want to render differently for each camera. 

I can see two solutions here:
1. Add `Camera` argument to the `render()` method.
2. Add static variable `Camera.current` and then have each camera set/clear that variable during its render call.

The first approach is more robust, but also more disruptive to existing user code. The second approach is more conservative but is essentially a hack to pass down the camera via unorthodox means.

Reply:
If the components are children of the camera, like in @luanpotter's tree, then we wouldn't need to do any of those solutions for `render` right?
We would have to make`update` only run once per component though, even though it can be a child of many `Camera`s.

Reply:
If we have a world that is observed through multiple cameras at the same time, then we can't have the world to be mounted to all those cameras. While it is superficially possible to `addChild(world)` on multiple cameras, it would result in `world` not having a proper notion of a parent, and further unknown consequences caused by the fact that our component tree is no longer a *tree*. 

A more sane approach would be for the camera to *know* about the world that it observes (via a property, similarly how in this PR the camera knows about the `ComponentSet` of its parent), but the world wouldn't actually be *mounted* to the camera. That way we wouldn't have to worry about duplicate `update()` calls either.

Note also that if we have a component within the world that cares about being observed, and there are multiple cameras looking at the world, then ultimately the component will have to be rendered differently even within the single render cycle.